### PR TITLE
Adapt `NetworkPolicy` rules according to recent refactorings

### PR DESCRIPTION
**Pull request by timuthy**
**How to categorize this PR?**

/area dev-productivity ops-productivity networking security
/kind enhancement

**What this PR does / why we need it**:
This PR adapts the `NetworkPolicy` rules according to the recent refactorings in `gardener/gardener`
See also https://github.com/gardener/gardener/pull/7589 and https://github.com/gardener/gardener/blob/master/docs/concepts/resource-manager.md#networkpolicy-controller.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/7352

**Special notes for your reviewer**:
/cc @rfranzke @oliver-goetz @ScheererJ

Depends on `gardener/gardener@v1.66` to create the expected network policies in the extension namespace.

**Release note**:

```breaking operator
When running this extension with Gardener `v1.66` and newer, several network policies are added to the extension namespace when feature gate `FullNetworkPoliciesInRuntimeCluster` is enabled. This can break existing setups if a **remote** DNS server is deployed to the **same cluster/network** as the central DNS controller. Earlier, egress connections for the DNS controller were unrestricted but with this change the controller will not be able to reach out to the remote DNS server in the same network.
If you run such a setup, please make sure to additionally deploy egress rules for the DNS controller in the extension namespace.
```
```other operator
Adapted extension components to support the [FullNetworkPoliciesInRuntimeCluster](https://github.com/gardener/gardener/blob/master/docs/deployment/feature_gates.md#list-of-feature-gates) feature gate introduced by `gardener/gardener` v1.66, see [here](https://github.com/gardener/gardener/blob/master/docs/concepts/resource-manager.md#networkpolicy-controller) and [#7352](https://github.com/gardener/gardener/pull/7589) for more information.
```

### CODE REVIEW

**Review by MartinWeindel — DISMISSED** [review]
/lgtm

As it is depending on gardener/gardener v1.66, should this fact not be reflected in the release notes?

**Comment by timuthy** [comment]
Thanks @MartinWeindel, I adjusted the release note.

File: charts/gardener-extension-shoot-dns-service/templates/dnsman-service-remoteaccess.yaml [hunk 1]
```diff
@@ -2,8 +2,9 @@
 apiVersion: v1
 kind: Service
 metadata:
-  {{- if .Values.dnsControllerManager.remoteaccess.service.annotations }}
   annotations:
+    networking.resources.gardener.cloud/from-world-to-ports: '[{"protocol":"TCP","port":{{ .Values.dnsControllerManager.remoteaccess.port }}}]'
```
Comment (rfranzke) [inline charts/gardener-extension-shoot-dns-service/templates/dnsman-service-remoteaccess.yaml:6]: When do we need this? Only when the service type is `LoadBalancer`?

Comment (timuthy) [reply charts/gardener-extension-shoot-dns-service/templates/dnsman-service-remoteaccess.yaml:8]: Yes, applied condition in [af9bd2f](https://github.com/gardener/gardener-extension-shoot-dns-service/pull/197/commits/af9bd2f5334416b515911616b08d6d2613274cb8).

**Review by rfranzke — APPROVED** [review]
/lgtm
/cc @MartinWeindel

**Pull request merged**